### PR TITLE
User/f1p/fix vmr

### LINKED ISSUE #228: Error in vmr adjustment in FV3
**Describe the bug**
FV3 fv_update_phys and fv_dynamics require tracers to be in mmr units. 
There is no adjustment for vmr tracers in fv_dynamics and the correction used in fv_update_phys is wrong. 

**To Reproduce**
Turn off all sources and sink of any vmr tracers. The burden is not conserved. 

**Expected behavior**
VMR global burden should be conserved by FV3.

**System Environment**
NOAA HPC

### PR DESCRIPTION
**Pull request by fabienpaulot**
**Description**

Implements fix to handle VMR tracers in FV3. Add new diagnostics for dry mmr and vmr and burden.
Caveat: This code assumes that tracers 1 to 3 are water tracers (in atmosphere.F90, mw_air_store).

Fixes #228 

**How Has This Been Tested?**

Tested in AM4.2 at GFDL 

**Checklist:**

Please check all whether they apply or not
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by bensonr** [comment]
@fabienpaulot - can you please point this at the dev/gfdl branch?

**Comment by fabienpaulot** [comment]
Should I create a new pull request @Rusty Benson - NOAA Federal
***@***.***> . I am not sure how to edit the existing one. Fabien

**Comment by fabienpaulot** [comment]
I think I figured it out. Sorry I am not good with git. Fabien

**Review by lharris4 — APPROVED** [review]
I think this looks good. I don't quite understand the purpose of the updates in the nudging, but the change to the VMR adjustment is good. The same idea can potentially also be done for tracers defined as dry-mass, as the folks at CSL are doing.

File: GFDL_tools/read_climate_nudge_data.F90 [hunk 1]
```diff
@@ -104,7 +104,7 @@ subroutine read_climate_nudge_data_init (nlon, nlat, nlev, ntime)
 !   ntime       number of time levels
 
   integer :: iunit, ierr, io
-  integer :: istat, i, j, k, n, nd, siz(4), i1, i2
```
[the remark this answers is not in the document]
Comment (uramirez8707) [reply GFDL_tools/read_climate_nudge_data.F90:107]: This fixed some of the logic in the nudging. This is already in the dev/gfdl:
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/180

**Comment by bensonr** [comment]
@lharris4 - there have been updates since you originally reviewed.  Could you perform another one - thanks

File: driver/GFDL/atmosphere.F90 [hunk 2]
```diff
@@ -582,8 +582,15 @@ subroutine atmosphere_dynamics ( Time, Surf_diff )
    if ( any((/ id_qdt_dyn, id_qldt_dyn, id_qidt_dyn, id_qadt_dyn /) > 0) .or. &
         query_cmip_diag_id(ID_tnhusa) ) qtend(:, :, :, 1:4) = Atm(mygrid)%q (isc:iec, jsc:jec, :, 1:4)
 
-   mw_air_store = Mw_air(sum(Atm(mygrid)%q(isc:iec,jsc:jec,:,1:3),dim=4)) !g/mol
+   mw_air_store = Mw_air(Atm(mygrid)%q (isc:iec, jsc:jec, :, sphum)   + &
```
Comment (lharris4) [inline driver/GFDL/atmosphere.F90:585]: Note that this may need to be revised if another microphysics scheme with more microphysical species is used.

**Review by lharris4 — APPROVED** [review]
Looks good. I believe the principal changes have been to the diagnostics, and an improved calculation of the variable mw_air_store?

**Comment by fabienpaulot** [comment]
I agree Lucas. Note that there are other places in FV3 that are hard-coded to use these 3 tracers for water so other changes would need to occur as well.
This is why I went that route.

**Comment by lharris4** [comment]
Hi, Fabien. Indeed we do want to go through and fix the use of 'nwat' to be
more flexible for new microphysics schemes. Thanks.

Lucas

**Review by uramirez8707 — APPROVED** [review]

**Pull request merged**